datafusion/sql/src/expr/function.rs

```diff
@@ -404,6 +404,11 @@ impl<S: ContextProvider> SqlToRel<'_, S> {
                     }
                     (!within_group.is_empty()).then_some(within_group)
                 } else {
+                    let order_by = if !order_by.is_empty() {
+                        order_by
+                    } else {
+                        within_group
```

Comment:
It should be disallowed to specify both order_by and within_group. A follow-up perhaps.